cmd/frontend/internal/auth/ipallowlist/checker/checker.go

```diff
@@ -0,0 +1,308 @@
+package checker
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/sourcegraph/log"
+
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/conf/conftypes"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+var (
+	// stableIpAllowlist is a list of IP addresses that are allowed to access
+	// permanetlly regardless of the configuration from site config.
+	//
+	// Use case:
+	//  - Cloud needs to configure IP allowlist that are managed by Cloud infra, and we don't user to override it, e.g., gke pod cidr, executors ip range.
+	stableUserIpAllowlist = func() []string {
+		if allowlist := os.Getenv("AUTH_ALLOWED_IP_ADDRESS_STABLE_USER_IP_ALLOWLIST"); allowlist != "" {
+			addr := strings.Split(allowlist, ",")
+			for i := range addr {
+				addr[i] = strings.TrimSpace(addr[i])
+			}
+			return addr
+		}
+		return nil
+	}()
+	stableClientIpAllowlist = func() []string {
+		if allowlist := os.Getenv("AUTH_ALLOWED_IP_ADDRESS_STABLE_CLIENT_IP_ALLOWLIST"); allowlist != "" {
+			addr := strings.Split(allowlist, ",")
+			for i := range addr {
+				addr[i] = strings.TrimSpace(addr[i])
+			}
+			return addr
+		}
+		return nil
+	}()
+
+	// configResolver is a cached resolver for the current configuration.
+	// If empty, implies it is disabled.
+	//
+	// It will attempt to parse all config with best effort, and
+	// attach a multi-error in the struct field `err` if any error occurred.
+	// The caller should decide what to do with the error.
+	configResolver = conf.Cached(getConfig)
+)
+
+func init() {
+	conf.ContributeValidator(validateSiteConfig)
+}
+
+// Checker implements IP allowlist checking and subscription to configuration changes.
+type Checker struct {
+	logger log.Logger
+}
+
+// config is the computed internal config from site config.
+// use configResolver() to get the current config.
+type config struct {
+	authorizedClientIps    []net.IP
+	authorizedClientRanges []net.IPNet
+	authorizedUserIps      []net.IP
+	authorizedUserRanges   []net.IPNet
+	userHeaders            []string
+	errorMessageTmpl       *template.Template
+
+	// err indicates error occurred while parsing the configuration.
+	err error
+}
+
+// New returns a new Checker. You must call Start() on the returned Checker
+// to subscribe to configuration changes in the background.
```

Comment:
I think there's no longer a `Start()`:

```suggestion
// New returns a new Checker.
```